pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -348,27 +362,32 @@ func (vm *windows) initializeBootstrapperFiles() error {
 // createService creates the service on the Windows VM
 func (vm *windows) createService(svc *service) error {
 	out, err := vm.Run("sc.exe create "+svc.name+" binPath=\""+svc.binaryPath+" "+
-		svc.args+"\" start=auto", false)
+		svc.args+" start=auto", false)
 	if err != nil {
 		return errors.Wrapf(err, "failed to create service with output: %s", out)
 	}
 	return nil
 }
 
 // stopService stops the service that was already running
-func (vm *windows) stopService(svc service) error {
-	out, err := vm.Run("sc.exe stop "+svc.Name(), false)
+func (vm *windows) stopService(svc *service) error {
+	out, err := vm.Run("sc.exe stop "+svc.name, false)
 	if err != nil {
-		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.Name(), out)
+		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.name, out)
 	}
 	return nil
 }
 
 // isRunning gets the status of given service
-func (vm *windows) isRunning(svc service) bool {
-	out, err := vm.Run("sc.exe query "+svc.Name(), false)
+func (vm *windows) isRunning(svc *service) bool {
```

Comment:
This function now does two things. Moreover it is returning `false` when the service does not exists which implies the service exists but is not running. This is incorrect semantics.

Reply:
split into two, that checks for existing first then checks if running.